src/components/CourseCard/BaseCourseCard.jsx

```diff
@@ -3,53 +3,55 @@ import PropTypes from 'prop-types';
 import classNames from 'classnames';
 import moment from 'moment';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-
-import './CourseCard.scss';
+import { StatusAlert } from '@edx/paragon';
 
 import EmailSettingsModal from './EmailSettingsModal';
 
+import './CourseCard.scss';
+
 class BaseCourseCard extends Component {
   state = {
     modals: {
-      email_settings: null,
+      emailSettings: null,
     },
+    hasEmailsEnabled: this.props.hasEmailSettings || false,
+    hasNewEmailSettings: false,
   };
 
   setModalState = ({ key, options }) => {
-    this.setState({
+    this.setState(state => ({
       modals: {
-        ...this.state.modals,
+        ...state.modals,
         [key]: options,
       },
-    });
+    }));
+  };
+
+  handleEmailSettingsModalOnClose = (hasEmailsEnabled) => {
+    this.resetModals();
+
+    if (hasEmailsEnabled !== undefined) {
+      this.setState({
+        hasNewEmailSettings: true,
+        hasEmailsEnabled,
+      });
+    }
   };
 
   resetModals = () => {
     this.setState({
       modals: {
-        email_settings: null,
+        emailSettings: null,
       },
     });
   };
 
-  handleEmailSettingsChange = () => {
```

Comment:
Removing this function because we don't actually have a need for doing anything in `BaseCourseCard` when the checkbox `checked` state changes.